crates/dbsp/src/trace/spine_async/mod.rs

```diff
@@ -534,8 +557,16 @@ where
 
         for (level, m) in mergers.iter_mut().enumerate() {
             if let Some(merger) = m.as_mut() {
-                let mut fuel = 10_000;
+                let starting_fuel = if level == 0 {
+                    isize::MAX
+                } else {
+                    stored_fuel[level].max(10_000)
+                };
+                let mut fuel = starting_fuel;
                 merger.work(&key_filter, &value_filter, &frontier, &mut fuel);
+                let fuel_consumed = starting_fuel - fuel;
+                stored_fuel[level] = (stored_fuel[level] - fuel_consumed).max(0);
+                stored_fuel[level + 1] += fuel_consumed;
```

Comment:
why do you add it to the next level?

Reply:
I was experimenting and it helped and @lalithsuresh wanted me to merge it without learning more.

Reply:
@blp Does line 569 need a bounds check for `levels + 1`? (not that we're ever at the last level)